### PR TITLE
Improve Jacoco code coverage

### PR DESCRIPTION
**Pull request by xiaoyu-yang-gh**
This is to improve the Jacoco code coverage:
(1) change the generate html report to make it more readable.
(2) include more subprojects to the report.
(3) exclude the generated code from the report.
(4) add unit tests for some zero covered tests.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -273,6 +273,13 @@ task codeCoverageReport(type: JacocoReport) {
           //add source sets from all relevant submodules
           sourceSets subproject.sourceSets.main
         }
+        classDirectories.setFrom(files(classDirectories.files.collect {
+          fileTree(dir: it, exclude: [
```
Comment (radai-rosenblatt) [inline build.gradle:277]: nit - can you add a comment explaining this is done to explode some classes from coverage by package?

File: common/build.gradle [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+ * Licensed under the BSD 2-Clause License (the "License").
+ * See License in the project root for license information.
+ */
+
+plugins {
+    id "java-library"
+    id "checkstyle"
+}
+
+dependencies {
```
Comment (radai-rosenblatt) [inline common/build.gradle:12]: nit - i think you can remove the entire dependencies {} section if there's nothing there ?

File: common/src/main/java/com/linkedin/avroutil1/common/ExcludeFromJacocoGeneratedReport.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+
+/**
+ * Annotation that can be used to exclude code from jacoco coverage.
+ */
+
+@Retention(RetentionPolicy.RUNTIME)
```
Comment (radai-rosenblatt) [inline common/src/main/java/com/linkedin/avroutil1/common/ExcludeFromJacocoGeneratedReport.java:20]: can we reduce this to SOURCE or something?

Comment (radai-rosenblatt) [reply common/src/main/java/com/linkedin/avroutil1/common/ExcludeFromJacocoGeneratedReport.java:22]: no, we cant

**Review by radai-rosenblatt — APPROVED** [review]

File: common/build.gradle (hunk 2 above)
Comment (xiaoyu-yang-gh) [reply common/build.gradle]: removed

File: build.gradle (hunk 1 above)
Comment (xiaoyu-yang-gh) [reply build.gradle:278]: added
